Nastya

asya-git opened the pull request:
1. Реализовать SingleTon класс Config. 
2. Конфигурацию приложение хранить в многомерном массиве в отдельном файле конфига.
3. При инициализации класса Config подгружать данные в свойство класса.
4. Данные о конфигурации должны быть приватными.
5. Получение данных о разделах конфигурации реализовать посредством создания соответствующих методов (прим. getDbConfig()).
6. Реализовать в классе Config возможность (метод) парсинга/получения данных из .env файла.
7. Реализацию SingleTon вынести в Трейт.
8.  Вывести данные конфига на главной странице в формате:
   `propertyGroup`:
         ‘propertyName’: ‘propertyValue

asya-git committed:
Create Singleton

asya-git committed:
Implement SingleTon

получать данные о разделах конфигурации
задавать данные для конфигов
получать/задавать данные из .env

IKolyas commented on Base/Config.php:
```diff
@@ -0,0 +1,77 @@
+    private ?array $env = null;
+
+
+    //static Config|null $instance = null;
```
убрать

IKolyas commented on Base/Config.php:
```diff
@@ -0,0 +1,77 @@
+        $this->dataDB = !empty($dbConfig) ? $dbConfig : null;
+
+    }
+    public function getConfigBD(): array|null
```
Рекомендация
?array

IKolyas commented on Base/Config.php:
```diff
@@ -0,0 +1,77 @@
+        $this->setConfigBD($this->config['db']);
+        $this->setConfigUser($this->config['user']);
+
+        $this->env = parse_ini_file('../.env');
```
Рекомендация
Проверка

IKolyas commented on Base/Config.php:
```diff
@@ -0,0 +1,77 @@
+    use SingleTon;
+
+    private ?array $dataDB = null;
+    private ?array $dataUser = null;
```
Упростить

IKolyas commented on Traits/SingleTon.php:
```diff
@@ -0,0 +1,15 @@
+namespace app\Traits;
+trait SingleTon 
+{
+    static object|null $instance = null;
```
private

IKolyas commented on Traits/SingleTon.php:
```diff
@@ -0,0 +1,15 @@
+{
+    static object|null $instance = null;
+
+    public static function getInstance():object
```
static

asya-git committed:
Fix bugs, added a method for changing the config

Ошибки: Чистка комментариев, проверка .env файла, упрощены конфиги
Добавлено: Изменение определенного конфига

asya-git committed:
fix bugs
